x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/textembedding/TextEmbeddingMlNodeService.java

```diff
@@ -52,46 +55,108 @@ public class TextEmbeddingService implements InferenceService {
 
     private final OriginSettingClient client;
 
-    public TextEmbeddingService(InferenceServiceExtension.InferenceServiceFactoryContext context) {
+    private static final Logger logger = LogManager.getLogger(TextEmbeddingMlNodeService.class);
+
+    public TextEmbeddingMlNodeService(InferenceServiceExtension.InferenceServiceFactoryContext context) {
         this.client = new OriginSettingClient(context.client(), ClientHelper.INFERENCE_ORIGIN);
     }
 
     @Override
-    public TextEmbeddingModel parseRequestConfig(
+    public void parseRequestConfig(
         String inferenceEntityId,
         TaskType taskType,
         Map<String, Object> config,
-        Set<String> platformArchitectures
+        Set<String> platformArchitectures,
+        ActionListener<Model> modelListener
     ) {
-        Map<String, Object> serviceSettingsMap = removeFromMapOrThrowIfNull(config, ModelConfigurations.SERVICE_SETTINGS);
+        try {
+            Map<String, Object> serviceSettingsMap = removeFromMapOrThrowIfNull(config, ModelConfigurations.SERVICE_SETTINGS);
+            String modelId = (String) serviceSettingsMap.get(MODEL_VERSION);
+            if (modelId == null) {
+                throw new IllegalArgumentException("Error parsing request config, model id is missing");
+            }
+            if (MultilingualE5SmallMlNodeServiceSettings.MODEL_VARIANTS.contains(modelId)) {
+                e5Case(inferenceEntityId, taskType, config, platformArchitectures, serviceSettingsMap, modelListener);
+            } else {
+                customElandCase(inferenceEntityId, taskType, config, platformArchitectures, serviceSettingsMap, modelListener);
+            }
+        } catch (Exception e) {
+            modelListener.onFailure(e);
+        }
+    }
 
-        var e5ServiceSettings = MultilingualE5SmallServiceSettings.fromMap(serviceSettingsMap);
+    private void customElandCase(
+        String inferenceEntityId,
+        TaskType taskType,
+        Map<String, Object> config,
+        Set<String> platformArchitectures,
+        Map<String, Object> serviceSettingsMap,
+        ActionListener<Model> modelListener
+    ) {
+        String modelId = (String) serviceSettingsMap.get(MODEL_VERSION);
+        var request = new GetTrainedModelsAction.Request(modelId);
+
+        var getModelsListener = modelListener.<GetTrainedModelsAction.Response>delegateFailureAndWrap((delegate, response) -> {
+            if (response.getResources().count() < 1) {
+                throw new IllegalArgumentException(
+                    "Error parsing request config, model id does not match any models available on this platform. Was ["
+                        + modelId
+                        + "]. You may need to load it into the cluster using eland."
```

Comment:
In practice this never happens because the GetTrainedModelsAction throws an exception when the model ID isn't found, but that is OK in this context (I updated that error message to mention loading the model with eland), and I'm leaving this code path here in case that behaviour ever changes.